build_tools/cmake/flatbuffer_cc_library.cmake

```diff
@@ -105,6 +105,11 @@ function(flatbuffer_cc_library)
     "" # copy_text_schemas_dir
   )
 
+  if(CMAKE_CROSSCOMPILING)
+    iree_get_host_exectuable_path(flatc _FLATC_PATH)
+    add_dependencies("${_NAME}_gen" iree_host_flatc)
```

Comment:
So this target should always be dependent on flatc. Instead of branching here, can we always use some function to access any binary dependency and that function deals with whether we're cross compiling?

Reply:
I normally won't worry that much if it's just for one case. I'd think about deduplication and abstraction when there are at least two cases. flatc dependency is the only third party code generator we are depending on in IREE for now. So I put it like that. But since you asked, I've created a function for this. :)